pallets/liquidity-rewards/src/lib.rs

```diff
@@ -55,92 +56,39 @@ pub use frame_support::{
 use frame_system::pallet_prelude::*;
 use num_traits::sign::Unsigned;
 pub use pallet::*;
-use sp_runtime::{
-	traits::{BlockNumberProvider, Saturating, Zero},
-	FixedPointOperand,
-};
+use sp_runtime::{traits::Zero, FixedPointOperand};
 use sp_std::mem;
 use weights::WeightInfo;
 
-/// Type that contains the timestamp of an epoch.
-#[derive(Encode, Decode, TypeInfo, MaxEncodedLen, RuntimeDebug)]
-#[cfg_attr(test, derive(PartialEq))]
-pub struct EpochTimestamp<BlockNumber>(BlockNumber);
-
-pub struct FirstEpochTimestamp<Provider, Duration>(
-	sp_std::marker::PhantomData<(Provider, Duration)>,
-);
-impl<Provider, Duration, BlockNumber> Get<EpochTimestamp<BlockNumber>>
-	for FirstEpochTimestamp<Provider, Duration>
-where
-	Provider: BlockNumberProvider<BlockNumber = BlockNumber>,
-	Duration: Get<BlockNumber>,
-	BlockNumber: Saturating,
-{
-	fn get() -> EpochTimestamp<BlockNumber> {
-		EpochTimestamp(Provider::current_block_number().saturating_add(Duration::get()))
-	}
-}
-
 /// Type that contains the associated data of an epoch
-#[derive(Encode, Decode, TypeInfo, MaxEncodedLen, RuntimeDebug)]
-#[cfg_attr(test, derive(PartialEq))]
-pub struct EpochData<BlockNumber, Balance, GroupId, Weight, MaxGroups>
-where
-	MaxGroups: Get<u32>,
-	GroupId: Ord,
-{
-	duration: BlockNumber,
-	reward: Balance,
-	weights: BoundedBTreeMap<GroupId, Weight, MaxGroups>,
+#[derive(Encode, Decode, TypeInfo, MaxEncodedLen, RuntimeDebugNoBound)]
+#[scale_info(skip_type_params(T))]
```

Comment:
A few days ago, I learned about this. It allows to use the `T: Config` generic in any type without notification to the type info that it exists, reducing a lot of generic boilerplate. For types like this, it works pretty fine.